users/views.py

```diff
@@ -2,52 +2,82 @@
 from django.contrib.auth.decorators import login_required
 from django.shortcuts import render, reverse
 from django.contrib.auth import authenticate, login, logout
-from users.forms import LoginForm
+from users.forms import LoginForm, UploadFileForm, UploadProfileImage
 from django.http import HttpResponseRedirect
 from users.admin import MyUserCreationForm
+from helpers.upload import handle_upload_file
 
 
-def handle_login(request):
-    if request.method == 'POST':
+def handle_login(request) :
+    if request.method == 'POST' :
         form = LoginForm(request.POST)
 
-        if form.is_valid():
+        if form.is_valid() :
             email = form.cleaned_data['email']
             password = form.cleaned_data['password']
             user = authenticate(request, username=email, password=password)
 
-            if user is not None:
+            if user is not None :
                 login(request, user)
                 return HttpResponseRedirect(reverse('users:profile'))
-    else:
+    else :
         form = LoginForm()
 
     return render(request, 'users/login.html', {
         'form' : form
     })
 
 
-def handle_logout(request):
+def handle_logout(request) :
     logout(request)
     return HttpResponseRedirect(reverse('users:login'))
 
 
 @login_required
-def profile(request):
-    return render(request, 'users/profile.html')
+def profile(request) :
+    if request.method == 'POST' :
+        user_profile = request.user.profile
+        form = UploadProfileImage(request.POST, request.FILES, instance=user_profile)
+
+        if form.is_valid() :
+            user_profile = form.save(commit=False)
+            user_profile.user = request.user
```

Comment:
Linia asta trebuie scoasa de aici. Ne asiguram folosind `post_save` de la ultimul curs (Week 10) ca cream in mod automat un profile atunci cand se salveaza prima data un user.
LE: In cazul asta putem salva direct profilul cu ajutorul form-ului, nu trebuie sa ii mai facem save manual.